scala/private/common.bzl

```diff
@@ -59,60 +30,62 @@ def _collect_jars_when_dependency_analyzer_is_off(
-            )
-
     return struct(
         compile_jars = depset(transitive = compile_jars),
         transitive_runtime_jars = depset(transitive = runtime_jars),
         jars2labels = JarsToLabelsInfo(jars_to_labels = jars2labels),
-        transitive_compile_jars = depset(transitive = compile_jars + plus_one_deps_compile_jars),
+        transitive_compile_jars = depset(transitive = transitive_compile_jars),
         deps_providers = deps_providers,
     )
 
-def _collect_jars_when_dependency_analyzer_is_on(dep_targets):
-    transitive_compile_jars = []
-    jars2labels = {}
-    compile_jars = []
-    runtime_jars = []
-    deps_providers = []
-
-    for dep_target in dep_targets:
-        # we require a JavaInfo for dependencies
-        # must use java_import or scala_import if you have raw files
-        java_provider = dep_target[JavaInfo]
-        deps_providers.append(java_provider)
-        current_dep_compile_jars = java_provider.compile_jars
-        current_dep_transitive_compile_jars = java_provider.transitive_compile_time_jars
-        runtime_jars.append(java_provider.transitive_runtime_jars)
-
-        compile_jars.append(current_dep_compile_jars)
-        transitive_compile_jars.append(current_dep_transitive_compile_jars)
+def collect_plugin_paths(plugins):
+    """Get the actual jar paths of plugins as a depset."""
+    paths = []
+    for p in plugins:
+        if hasattr(p, "path"):
+            paths.append(p)
+        elif JavaInfo in p:
+            paths.extend([j.class_jar for j in p[JavaInfo].outputs.jars])
+            # support http_file pointed at a jar. http_jar uses ijar,
+            # which breaks scala macros
 
-        add_labels_of_jars_to(
-            jars2labels,
-            dep_target,
-            current_dep_transitive_compile_jars.to_list(),
-            current_dep_compile_jars.to_list(),
-        )
+        elif hasattr(p, "files"):
+            paths.extend([f for f in p.files.to_list() if not_sources_jar(f.basename)])
+    return depset(paths)
 
-    return struct(
-        compile_jars = depset(transitive = compile_jars),
-        transitive_runtime_jars = depset(transitive = runtime_jars),
-        jars2labels = JarsToLabelsInfo(jars_to_labels = jars2labels),
-        transitive_compile_jars = depset(transitive = transitive_compile_jars),
-        deps_providers = deps_providers,
-    )
+def _additional_transitive_compile_jars(
+        java_provider,
+        dep_target,
+        dependency_mode):
+    if dependency_mode == "transitive":
+        return java_provider.transitive_compile_time_jars
+    elif dependency_mode == "plus-one":
+        if PlusOneDeps in dep_target:
+            plus_one_jars = [dep[JavaInfo].compile_jars for dep in dep_target[PlusOneDeps].direct_deps if JavaInfo in dep]
+            return depset(transitive = plus_one_jars + [java_provider.compile_jars])
+        else:
+            return java_provider.compile_jars
```

Comment:
I know this is my logic but I don't fully remember the cause. Probably some dependencies that the aspect didn't reach or something but I can't remember which or why.
Any chance you can try to remove it, see which tests fail and add a comment to the code? It's a complicated piece of code (and not due to your fault)

Reply:
Ok will play around and see

Reply:
Ok so the thing to determine is on the line `return depset(transitive = plus_one_jars + [java_provider.compile_jars])`, why do we need `[java_provider.compile_jars]`?

Reply:
Why is that the thing to determine? I don't remember the specific motivation but I can definitely tell you that a clear test will fail if you will remove.
I meant the `else` for ` if PlusOneDeps in dep_target:`

Reply:
Hmm okay have found some stuff and will add comments wrt that, hopefully it will make sense

Reply:
❤️ thanks!